src/components/useCurrencyConverter.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+
+export default function useCurrencyConverter(endpoint, access_key) {
+  const [currencyOptions, setCurrencyOptions] = useState([]);
+  const [fromCurrency, setFromCurrency] = useState();
+  const [toCurrency, setToCurrency] = useState();
+  const [exchangeRate, setExchangeRate] = useState();
+  const [amount, setAmount] = useState(1);
+  const [isAmountInFromCurrency, setAmountInFromCurrency] = useState(true);
+  const [currencies, setCurrencies] = useState({});
+  const [timestamp, setTimestamp] = useState();
+
+  useEffect(() => {
+    fetch(
+      `http://api.exchangeratesapi.io/v1/${endpoint}?access_key=${access_key}`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setCurrencies(data);
+        setTimestamp(data.timestamp);
+        const initialCurrency = Object.keys(data.rates)[0];
+        setCurrencyOptions([data.base, ...Object.keys(data.rates)]);
+        setFromCurrency(data.base);
+        setToCurrency(initialCurrency);
+        setExchangeRate(data.rates[initialCurrency]);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (fromCurrency != null && toCurrency != null && currencies != null) {
+      const rate = calculateExchangeRate(fromCurrency, toCurrency);
+      if (rate !== null) {
+        setExchangeRate(rate);
+      }
+    }
+  }, [fromCurrency, toCurrency, currencies]);
+
+  function calculateExchangeRate(fromCurrency, toCurrency) {
+    if (fromCurrency === currencies.base) {
+      return currencies.rates[toCurrency];
+    } else if (toCurrency === currencies.base) {
+      return 1 / currencies.rates[fromCurrency];
+    } else {
+      return currencies.rates[toCurrency] / currencies.rates[fromCurrency];
+    }
+  }
```

Comment:
When possible, keep function declarations out of component bodies. 

Everything inside a react component is recreated on each render unless memoized with useMemo or useCallback.

I would suggest to either
1. move the function out of the component and add the `currencies` as argument
```js
function calculateExchangeRate(currencies, from, to) {
  ... // rest can be the same
}
```
2. memoize it with useCallback and keep inside the component body
```js
const calculateExchangeRate = useCallback((from, to) => {
  ... // rest can be the same
}, [currencies])
```
But a rule of thumb is to keep the functions out of components when possible, so I would prefer the first option.

Keeping it out the component body also enhances readability.